server/routes/referrals/furtherInformationPresenter.ts

```diff
@@ -1,17 +1,35 @@
-import { ServiceCategory } from '../../services/interventionsService'
+import { DraftReferral, ServiceCategory } from '../../services/interventionsService'
 
 export interface FurtherInformationError {
   message: string
 }
 
 export default class FurtherInformationPresenter {
   constructor(
+    private readonly referral: DraftReferral,
     private readonly serviceCategory: ServiceCategory,
-    readonly error: FurtherInformationError | null = null
+    readonly error: FurtherInformationError | null = null,
+    private readonly userInputData: Record<string, string> | null = null
   ) {}
 
   readonly title = `Do you have further information for the ${this.serviceCategory.name} service provider? (optional)`
 
   readonly hint =
     'For example, relevant previous offences, previously completed programmes or further reasons for this referral'
+
+  get value(): string {
+    if (this.hasUserInputOrEmptyFormSubmitted) {
+      return this.userInputData!['further-information']
+    }
+
+    if (this.referral.furtherInformation) {
+      return this.referral.furtherInformation
+    }
+
+    return ''
+  }
+
+  private get hasUserInputOrEmptyFormSubmitted(): boolean {
+    return this.userInputData !== null && this.userInputData['further-information'] !== undefined
```

Comment:
Similar comment to [the one](https://github.com/ministryofjustice/hmpps-interventions-ui/pull/51#discussion_r551248907) I left on #51 - I don't think we want to check for the presence of that particular field. All we care about is - did the user submit any data (i.e. are we displaying this in response to a POST request)? If so, then replay whatever data they entered, if anything.

Also, in #52 I [encapsulated a bunch of this logic in a helper class](https://github.com/ministryofjustice/hmpps-interventions-ui/pull/54/files#diff-5f63ac2d1f262adf16ff47612e788fd4c885d691208a1e509bc10a71548c9dd4R66). Once that's merged it would be good to bring this in line. I'm hoping we can simplify our tests then too.

Reply:
Addressed the issue here, and will update to use helper class once #54 merged.